edweiss-app/app/(app)/quiz/createQuizPage.tsx

```diff
@@ -0,0 +1,612 @@
+						</>)
+					}
+				}
+				</For>
+
+			</TScrollView>
+			<FancyButton icon='add' backgroundColor='blue' onPress={() => addModalRef.current?.present()} ></FancyButton>
+			<FancyButton onPress={() => { checkQuizParams() ? publishQuizModalRef.current?.present() : {} }}> Publish quiz </FancyButton>
+			<AddExerciseModal modalRef={addModalRef} updateExerciseList={addToExerciseList} />
+			<PublishQuizModal modalRef={publishQuizModalRef} quizName={quizName} dueDate={dueDate == undefined ? "undefined" : dueDate.toDateString()} numberOfExercises={exercises.length} publishQuiz={publishQuiz} />
+
+		</>
+
+	);
+};
+export default CreateQuizPage;
+
+function timestampToDate(timestamp: number): Date {
+	if (typeof timestamp !== "number" || isNaN(timestamp)) {
+		throw new Error("Invalid timestamp. It must be a valid number.");
+	}
+
+	return new Date(timestamp);
+}
+
+export const NameAndDateCustom: ReactComponent<{ dueDate: Date | undefined, setQuizName: React.Dispatch<React.SetStateAction<string>>, showDate: boolean, setShowDate: React.Dispatch<React.SetStateAction<boolean>>, showTime: boolean, setShowTime: React.Dispatch<React.SetStateAction<boolean>>, onChangeDate: (date: Date | undefined) => void, onChangeTime: (date: Date | undefined) => void }> = ({ dueDate, setQuizName, showDate, setShowDate, showTime, setShowTime, onChangeDate, onChangeTime }) => {
+	return (<>
+		<RouteHeader title='Create a new quiz' />
+		<FancyTextInput mb='md' label='Quiz name' onChangeText={setQuizName}></FancyTextInput>
+		<TText> Due date : {dueDate == undefined ? "undefined" : dateToStringWithTime(dueDate)} { }</TText>
+
+		<FancyButton mb='md' onPress={() => setShowDate(true)}> Change due date </FancyButton>
+		<FancyButton mb='md' onPress={() => setShowTime(true)}> Change time </FancyButton>
+
+
+		{
+			showDate && (
+				<DateTimePicker
+					testID="dateTimePicker1"
+					value={dueDate == undefined ? timestampToDate(Date.now()) : dueDate}
+					mode='date'
+					is24Hour={true}
+					display="default"
+					onChange={(event, selectedDate) => {
+						if (event.type === "dismissed") {
+							setShowDate(false);
+						} else {
+							onChangeDate(selectedDate);
+
+						}
+					}
+					}
+				/>
+			)
+		}
+
+		{
+			showTime && (
+				<DateTimePicker
+					testID="dateTimePicker2"
+					value={dueDate == undefined ? timestampToDate(Date.now()) : dueDate}
+					mode='time'
+					is24Hour={true}
+					display="default"
+					onChange={(event, selectedDate) => {
+						if (event.type === "dismissed") {
+							setShowTime(false);
+						} else {
+							onChangeTime(selectedDate);
+						}
+					}}
+				/>
+			)
+		}
+	</>
+
+	);
+};
+
+function dateToStringWithTime(date: Date | undefined): string {
+	if (date == undefined) {
+		return "undefined"
+	}
+	if (!(date instanceof Date) || isNaN(date.getTime())) {
+		throw new Error("Invalid Date instance.");
```

Comment:
You can put yours errors messages together by exemple : 

```
const ERROR_MESSAGES = {
    invalid_date: "Invalid Date instance.",
    empty_question: "Question cannot be empty.",
    max_propositions: "You can add a maximum of 6 propositions.",
    // ...
};

```

Reply:
Do you mean in the quiz.json file?